zebra-chain/src/orchard/note/ciphertexts.rs

```diff
@@ -0,0 +1,133 @@
+use std::{fmt, io};
+
+use crate::serialization::{serde_helpers, SerializationError, ZcashDeserialize, ZcashSerialize};
+
+/// A ciphertext component for encrypted output notes.
+///
+/// Corresponds to the Orcahrd 'encCiphertext's
+#[derive(Deserialize, Serialize)]
+pub struct EncryptedNote(#[serde(with = "serde_helpers::BigArray")] pub [u8; 580]);
+
+impl fmt::Debug for EncryptedNote {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_tuple("EncryptedNote")
+            .field(&hex::encode(&self.0[..]))
+            .finish()
+    }
+}
+
+// These impls all only exist because of array length restrictions.
+
+impl Copy for EncryptedNote {}
+
+impl Clone for EncryptedNote {
+    fn clone(&self) -> Self {
+        let mut bytes = [0; 580];
+        bytes[..].copy_from_slice(&self.0[..]);
+        Self(bytes)
+    }
+}
+
+impl PartialEq for EncryptedNote {
+    fn eq(&self, other: &Self) -> bool {
+        self.0[..] == other.0[..]
+    }
+}
+
+impl Eq for EncryptedNote {}
+
+impl ZcashSerialize for EncryptedNote {
+    fn zcash_serialize<W: io::Write>(&self, mut writer: W) -> Result<(), io::Error> {
+        writer.write_all(&self.0[..])?;
+        Ok(())
+    }
+}
+
+impl ZcashDeserialize for EncryptedNote {
+    fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
+        let mut bytes = [0; 580];
+        reader.read_exact(&mut bytes[..])?;
+        Ok(Self(bytes))
+    }
+}
+
+/// A ciphertext component for encrypted output notes.
+///
+/// Corresponds to Orcahrd's 'outCiphertext'
```

Comment:
There's a few of these typos, can you do a search and replace?

```suggestion
/// Corresponds to Orchard's 'outCiphertext'
```